AccountingSystem/AccountingSystem/GasStation.cpp

```diff
@@ -0,0 +1,92 @@
+#include <sstream>
+#include <string>
+#include <vector>
+#include <ctime>
+#include <iomanip>
+#include "GasStation.h"
+
+GasStation::GasStation(IDataHandler<GasStationData>* handler):_handler(handler)
+{
+}
+
+void GasStation::AddRecord(InputData userData,const bool IncreaseTheBallance)
+{
+    GasStationData data;
+    data.Name = userData.Name;
+    data.Surname = userData.Surname;
+    data.Liters = userData.Liters;
+    data.ID = GenerateNextId();
+    data.Date = GetCurrentDate();
+    auto ballance = GetCurrentBallance();
+    data.Ballance = IncreaseTheBallance ? ballance + data.Liters : ballance - data.Liters;
```

Comment:
User will supply negative `Litres` if he wants to decrease the balance

```suggestion
    data.Ballance = ballance + data.Liters;
```